### ISSUE TITLE
Add support for mark - highlighting words

### ISSUE THREAD

**Issue by weilinzung** [markedjs/marked#2563]
**Describe the feature**
https://www.markdownguide.org/extended-syntax/#highlight

Also, related to type as well if this added: https://www.npmjs.com/package/@types/marked

**Why is this feature necessary?**
```
I need to highlight these ==very important words==.
```

```
I need to highlight these <mark>very important words</mark>.
```

**Comment by UziTech** [comment]
You can create an extension for this:

```js
const markHighlight = {
  name: 'markHighlight',
  level: 'inline',
  start(src) { return src.indexOf('=='); },   // Hint to Marked.js to stop and check for a match
  tokenizer(src, tokens) {
    const rule = /^==(.*?)==/;                // Regex for the complete token, anchor to string start
    const match = rule.exec(src);
    if (match) {
      return {                                // Token to generate
        type: 'markHighlight',                // Should match "name" above
        raw: match[0],                        // Text to consume from the source
        tokens: this.lexer.inline(match[1]),  // any further-nested inline tokens
      };
    }
  },
  renderer(token) {
    return `<mark>${this.parser.parseInline(token.tokens)}</mark>`;
  },
};

marked.use({ extensions: [markHighlight] });

console.log(marked.parse('==highlighted text=='));
// <p><mark>highlighted text</mark></p>
```

**Comment by rvanzon** [comment]
@UziTech thanks for your help! Though, the inline tokens don't seem to get fetched and result in undefined tokens.

Changing it to this works:

```javascript
    if (match) {
      const token = {
        // Token to generate
        type: "markHighlight", // Should match "name" above
        raw: match[0], // Text to consume from the source
        tokens: [],
      };
      this.lexer.inline(match[1], token.tokens); // any further-nested inline tokens
      return token;
    }
```

**Comment by UziTech** [comment]
That should've been fixed in [v4.0.19](https://github.com/markedjs/marked/releases/tag/v4.0.19). My code should work in the latest version.

**Comment by rvanzon** [comment]
@UziTech ah, didn't know, I will upgrade. Thanks again!
